src/legal-officer/transaction-protection/LocPublicDataButton.tsx

```diff
@@ -6,25 +6,33 @@ import { useForm } from "react-hook-form";
 import React, { useCallback, useState } from "react";
 import { useLocContext } from "./LocContext";
 import Icon from "../../common/Icon";
+import { LocItem } from "./types";
 
 export default function LocPublicDataButton() {
 
     const { colorTheme } = useCommonContext();
     const [ visible, setVisible ] = useState(false);
     const { control, handleSubmit, formState: { errors }, reset } = useForm<FormValues>();
-    const { addMetadata } = useLocContext();
+    const { addMetadata, locItems } = useLocContext();
+    const [ existingItem, setExistingItem ] = useState<LocItem | undefined>(undefined);
 
-    const submit = useCallback((formValues: FormValues) => {
-        addMetadata!(formValues.dataName, formValues.dataValue)
-        setVisible(false)
-    }, [ addMetadata ]);
+    const submit = useCallback(async (formValues: FormValues) => {
```

Comment:
Useless "async", there is no await in the function's body.

Reply:
async removed.